src/ui.rs

```diff
@@ -64,14 +64,21 @@ pub fn update_health_bar(
     heal_bar.width = Val::Percent((health.0 / MAX_PLAYER_HEALTH) as f32 * 100.);
 }
 
+#[derive(PartialEq, Component, Debug)]
+pub enum SpellType {
+    Fireball,
+    None,
+}
+
 #[derive(Component)]
-struct InventorySlot {
-    index: usize,
+pub struct InventorySlot {
+    pub index: usize,
+    pub spell: SpellType,
```

Comment:
should just be `Option<SpellType>`

and SpellType should not have a None variant...